bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/Workbench.java

```diff
@@ -510,6 +519,19 @@ public void eventLoopException(Throwable exception) {
 				new WorkbenchLocationService(IServiceScopes.WORKBENCH_SCOPE, this, null, null, null, null, 0));
 	}
 
+	private static void setEdgeDataDirectory(Display display) {
+		Location workspaceLocation = Platform.getInstanceLocation();
+		if (workspaceLocation != null) {
+			URI workspaceLocationURI;
+			try {
+				workspaceLocationURI = workspaceLocation.getURL().toURI();
+				display.setData(EDGE_USER_DATA_FOLDER, Paths.get(workspaceLocationURI).toString());
```

Comment:
I assume that `Platform.getInstanceLocation()` never returns a _relative_ Location (if that is even a thing). If that's the case then this code is fine, if that is **not** the case and therefore `workspaceLocationURI` may be _relative_ then adding a call to `Path::toAbsolutePath` would be necessary.

Reply:
The location will always be absolute. When initializing the workspace, the path will always be made absolute. When initially passing some relative location, it will be resolve relative to the install directory upon workspace initialization.

Reply:
Thanks for checking it!